partcad/src/partcad/shape.py

```
@@ -166,6 +166,16 @@ def get_cache_dependencies_broken(self) -> bool:
     def get_cacheable(self) -> bool:
         return self.cacheable and not self.get_cache_dependencies_broken()
 
+    async def get_summary_async(self, project=None):
+        # Return a manually configured summary if present; otherwise None so
+        # that AI-backed shapes (ShapeWithAi) can generate one via super().
+        if "summary" in self.config and not self.config["summary"] is None:
```

Ruff (0.15.21)

[error] Test for object identity should be `is not`

Convert to `is not`

(E714)